src/util.js

```diff
@@ -67,5 +66,24 @@ exports.certificateForKey = (key, privateKey, callback) => {
   // self-sign certificate
   cert.sign(privateKey)
 
-  return callback(null, cert)
+  return cert
 }
+
+/**
+ * Finds the first item in a collection that is matched in the
+ * `asyncCompare` function.
+ *
+ * `asyncCompare` is an async function that must
+ * resolve to either `true` or `false`.
+ *
+ * @param {Array} array
+ * @param {function(*)} asyncCompare An async function that returns a boolean
+ */
+async function findAsync (array, asyncCompare) {
+  const promises = array.map(asyncCompare)
+  const results = await Promise.all(promises)
```

Comment:
We have to wait for all these promises to resolve before we can find the value. This is a change in behaviour from `async/detect` which executes in parallel but will call the callback _as soon as _ one returns true.

...we can leave this for a separate PR.